conformity/resource_conformity_custom_rule.go

```diff
@@ -349,7 +358,17 @@ func processInputCustomRuleConditions(conditionsIn []interface{}) []cloudconform
 		obj := cloudconformity.CustomRuleCondition{}
 		obj.Fact = m["fact"].(string)
 		obj.Path = m["path"].(string)
-		obj.Value = m["value"].(string)
+		objValue := ObjectValue{}
+		if strings.ToLower(m["value"].(string)) == "true" || strings.ToLower(m["value"].(string)) == "false" {
+			obj.Value, _ = strconv.ParseBool(m["value"].(string))
+		} else if numValue, err := strconv.Atoi(m["value"].(string)); err == nil {
+			obj.Value = numValue
+		} else if err := json.Unmarshal([]byte(m["value"].(string)), &objValue); err == nil {
+			obj.Value = objValue
+		} else {
+			obj.Value = m["value"]
+		}
```

Comment:
⛏️ Comment here will help future developers
```suggestion
                 // Allow value to be boolean, number, object or string
                 // boolean, number and object expect user to use jsonencode function to encode (see docs) 
		objValue := ObjectValue{}
		if strings.ToLower(m["value"].(string)) == "true" || strings.ToLower(m["value"].(string)) == "false" {
			obj.Value, _ = strconv.ParseBool(m["value"].(string))
		} else if numValue, err := strconv.Atoi(m["value"].(string)); err == nil {
			obj.Value = numValue
		} else if err := json.Unmarshal([]byte(m["value"].(string)), &objValue); err == nil {
			obj.Value = objValue
		} else {
			obj.Value = m["value"]
		}
```